Misc/NEWS.d/next/Library/2024-01-14-11-43-31.gh-issue-113878.dmEIN3.rst

```diff
@@ -0,0 +1,6 @@
+Add ``doc`` parameter to :func:`dataclasses.field`, so it can be stored and
+shown as a documentation / metadata.
+It is only visible when ``@dataclass(slots=True)`` is used.
+
+In order to support this feature we are changing the ``__slots__`` format
+in dataclasses from :class:`tuple` to :class:`dict`.
```

Comment:
I was thinking of something like:

Add ``doc`` parameter to :func:`dataclasses.field`, so it can be stored and
shown as a documentation / metadata.  If ``@dataclass(slots=True)`` is used, 
then the supplied string is availabl in the ``__slots__`` dict.  Otherwise,
the supplied string is only available in the corresponding ``Field`` object.

In order to support this feature we are changing the ``__slots__`` format
in dataclasses from :class:`tuple` to :class:`dict`.